server.js

```diff
@@ -0,0 +1,39 @@
+const http = require("http");
+const url = require("url");
+const fs = require("fs");
+
+const wwwRoot = __dirname + '/www/';
+console.log(wwwRoot);
+
+const host = 'localhost';
+const port = 8000;
+
+const requestListener = function (req, res) {
+
+    console.log(`url : ${req.url}`);
+    const url_parts = url.parse(req.url);
+
+    // map request url to filesystem:
+    if (url_parts.pathname === '/' || url_parts.pathname === '/index.html') {
+        console.log('www root index');
+        fs.readFile(wwwRoot + 'index.html', 'UTF-8', (err, data) => {
```

Comment:
A co jeśli nie będzie pliku `index.html`?

Reply:
w następnej linijce jest `if(err) .... `
a niżej funkcja którą kopnie - error404()

btw. da się w tym miejscu  (odpowiedź na komentarz) wstawić dalszy kod?

Reply:
Na pewno da się w komentarzu wkleić jakiś kod, jeśli o to pytasz, ale komentarze są pod konkretnymi liniami i one "rozdzielają" kod

Reply:
A wracając do kodu - jak nie będzie `index.html` to będzie listing plików, czy nie bardzo? Na razie pytam, bo potem i tak będę musiał odpalić i sprawdzić :)

Reply:
no to na razie jest ogarnięty tylko www root dir i jak nie ma index.html to leci 404, ale jest juz pomysl jak to ruszyć dalej z podfolderami i pozostalymi rzeczami. Looknij na dziennik bo tam mialem pytanie co masz na mysli jako listing pliku i jak bardzo sie do tego przywiazujemy - czy sprawdzamy mime, ustawiamy content-type czy tylko jakies glowne typy a reszte oleawmy